site-src/v1alpha2/guides/http-redirect-rewrite.md

```diff
@@ -0,0 +1,180 @@
+
+## Redirects
+
+Redirects return HTTP 3XX responses to a client, instructing it to retrive a
+different resource. [`RequestRedirect` rule
+filters](v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRequestRedirectFilter)
+instruct Gateways to emit a redirect response to requests matching a filtered
+HTTPRoute rule.
+
+Redirect filters can substitute various URL components independently. For
+example, to issue a permanent redirect (301) from HTTP to HTTPS, configure
+`requestRedirect.statusCode=301` and `requestRedirect.scheme="https"`:
+
+```
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-filter-redirect
+spec:
+  hostnames:
+    - redirect.example
+  rules:
+    - filters:
+      - type: RequestRedirect
+	    requestRedirect:
+          scheme: https
+		  statusCode: 301
+      backendRefs:
+      - name: example-svc
+        weight: 1
+        port: 80
+```
+
+Redirects change configured URL components to match the redirect configuration
+while preserving other components from the original request URL. In this
+example, the request `GET http://redirect.example/cinammon` will result in a
+301 response with a `location: https://redirect.example/cinnamon` header. The
+hostname (`redirect.example`), path (`/cinnamon`), and port (implicit) remain
+unchanged.
+
+<!---
+The above raises some questions. Do redirects apply unconditionally? If my
+request is already HTTPS, will the above create a redirect loop? If redirects
+are unconditional (i.e. they _do not_ nothing and proxy upstream if the request
+already uses HTTPS), how should you properly write configuration elsewhere to
+ensure this only applies to HTTP requests?
+-->
+
+### Path redirects
+
+Path redirects use an HTTP Path Modifier to replace either entire paths or path
+prefixes. For example, 
+
+```
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-filter-redirect
+spec:
+  hostnames:
+    - redirect.example
+  rules:
+    - matches:
+      - path:
+          type: PathPrefix
+          value: /cayenne
+      filters:
+      - type: RequestRedirect
+	    requestRedirect:
+          path:
+		    type: ReplaceFullPath
+			replaceFullPath: /paprika
+		  statusCode: 302
+      backendRefs:
+      - name: example-svc
+        weight: 1
+        port: 80
+```
+
+will issue a 302 redirect to all `redirect.example` requests whose path begins
+with `/cayenne` to `/paprika`. Both requests to
+`https://redirect.example/cayenne/pinch` and
+`https://redirect.example/cayenne/teaspoon` will receive a redirect with a
+`location: https://redirect.example/paprika`.
+
+The other path redirect type, `ReplacePrefixMatch`, replaces only the path
+portion matching `matches.path.value`. Changing the filter in the above to:
+
+```
+      - type: RequestRedirect
+	    requestRedirect:
+          path:
+		    type: ReplacePrefixMatch
+			replacePrefixMatch: /paprika
+		  statusCode: 302
+```
```

Comment:
Indentation looks off here, maybe a combination of tabs and spaces going on here?